metis-enrichment/metis-enrichment-common/src/main/java/eu/europeana/enrichment/utils/EntityMergeUtils.java

```diff
@@ -0,0 +1,899 @@
+import eu.europeana.corelib.definitions.jibx.InScheme;
+import eu.europeana.corelib.definitions.jibx.IsPartOf;
+import eu.europeana.corelib.definitions.jibx.IsRelatedTo;
+import eu.europeana.corelib.definitions.jibx.Lat;
+import eu.europeana.corelib.definitions.jibx.LiteralType;
+import eu.europeana.corelib.definitions.jibx.NarrowMatch;
+import eu.europeana.corelib.definitions.jibx.Narrower;
+import eu.europeana.corelib.definitions.jibx.Notation;
+import eu.europeana.corelib.definitions.jibx.Note;
+import eu.europeana.corelib.definitions.jibx.PlaceType;
+import eu.europeana.corelib.definitions.jibx.PrefLabel;
+import eu.europeana.corelib.definitions.jibx.ProfessionOrOccupation;
+import eu.europeana.corelib.definitions.jibx.ProxyType;
+import eu.europeana.corelib.definitions.jibx.RDF;
+import eu.europeana.corelib.definitions.jibx.Related;
+import eu.europeana.corelib.definitions.jibx.RelatedMatch;
+import eu.europeana.corelib.definitions.jibx.ResourceOrLiteralType;
+import eu.europeana.corelib.definitions.jibx.SameAs;
+import eu.europeana.corelib.definitions.jibx.TimeSpanType;
+import eu.europeana.corelib.definitions.jibx._Long;
+import eu.europeana.enrichment.api.external.model.Agent;
+import eu.europeana.enrichment.api.external.model.EnrichmentBase;
+import eu.europeana.enrichment.api.external.model.Label;
+import eu.europeana.enrichment.api.external.model.LabelResource;
+import eu.europeana.enrichment.api.external.model.Part;
+import eu.europeana.enrichment.api.external.model.Place;
+import eu.europeana.enrichment.api.external.model.Resource;
+import eu.europeana.enrichment.api.external.model.Timespan;
+import eu.europeana.corelib.definitions.jibx.Concept.Choice;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.jibx.runtime.BindingDirectory;
+import org.jibx.runtime.IBindingFactory;
+import org.jibx.runtime.IUnmarshallingContext;
+import org.jibx.runtime.JiBXException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Created by erikkonijnenburg on 28/07/2017.
+ */
+public class EntityMergeUtils {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(EntityMergeUtils.class);
+  private final static String UTF8= "UTF-8";
+  private static IBindingFactory agentFactory;
+  private static IBindingFactory conceptFactory;
+  private static IBindingFactory placeFactory;
+  private static IBindingFactory tsFactory;
+  private static IBindingFactory rdfFactory;
+
+  static {
+    try {
+      agentFactory = BindingDirectory.getFactory(AgentType.class);
+      conceptFactory = BindingDirectory.getFactory(Concept.class);
+      placeFactory = BindingDirectory.getFactory(PlaceType.class);
+      tsFactory = BindingDirectory.getFactory(TimeSpanType.class);
+      rdfFactory = BindingDirectory.getFactory(RDF.class);
+    } catch (JiBXException e) {
+      LOGGER.error("Unable to get BindingFactory", e);
+      System.exit(-1);
```

Comment:
This seems a bit dramatic and dangerous (this class is to be used as a library, a hard exit can have unexpected consequences). Maybe we are better off with some other method of checking for errors.

Reply:
The System.exit is called in the static initializer block and as such I don't see a problem with this. Since this is a crucial class, the classloader should be able to load it without an error/exception and if there is an error, stopping is not a crazy option. This static block should never be catching an exception and then letting the application continue.

Reply:
I fixed it.